### ISSUE TITLE
Getting IndexOutOfBoundsException

### ISSUE THREAD

**Issue by elikohen** [splitwise/TokenAutoComplete#168]
We got some IndexOutOfBoundsException, here's the info:
- Lib version: 2.0
- Device: NVidia Shield Tablet
- Android version: 5.1.1
- Our application: Ztory

Here's the stacktrace: 
```
java.lang.IndexOutOfBoundsException: charAt: 0 >= length 0
       at android.text.SpannableStringBuilder.charAt(SpannableStringBuilder.java:116)
       at com.tokenautocomplete.TokenCompleteTextView$TokenTextWatcher.onTextChanged(TokenCompleteTextView.java:1164)
       at android.widget.TextView.sendOnTextChanged(TextView.java:7682)
       at android.widget.TextView.handleTextChanged(TextView.java:7742)
       at android.widget.TextView$ChangeWatcher.onTextChanged(TextView.java:9475)
       at android.text.SpannableStringBuilder.sendTextChanged(SpannableStringBuilder.java:964)
       at android.text.SpannableStringBuilder.replace(SpannableStringBuilder.java:515)
       at android.text.SpannableStringBuilder.delete(SpannableStringBuilder.java:216)
       at android.text.SpannableStringBuilder.delete(SpannableStringBuilder.java:33)
       at android.view.inputmethod.BaseInputConnection.deleteSurroundingText(BaseInputConnection.java:244)
       at android.view.inputmethod.InputConnectionWrapper.deleteSurroundingText(InputConnectionWrapper.java:66)
       at com.tokenautocomplete.TokenCompleteTextView$TokenInputConnection.deleteSurroundingText(TokenCompleteTextView.java:1333)
       at com.android.internal.view.IInputConnectionWrapper.executeMessage(IInputConnectionWrapper.java:389)
       at com.android.internal.view.IInputConnectionWrapper$MyHandler.handleMessage(IInputConnectionWrapper.java:78)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:135)
       at android.app.ActivityThread.main(ActivityThread.java:5273)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
```

**Comment by mgod** [comment]
I'm seeing occasional crashes like this as well. Do you have any idea how to reproduce this?

**Comment by mgod** [comment]
might be a dup of #160

**Comment by elikohen** [comment]
We just got it on crashlytics and just to one user but never happened while developing. The stack trace on #160 seems identical. I'll try to reproduce on Monday at the office. Thanks for the reply.

**Comment by dkadrikj** [comment]
Same here 2 times. Also got 2 times, library version 1.3.3 

>java.lang.IndexOutOfBoundsException: charAt: 0 >= length 0
       at android.text.SpannableStringBuilder.charAt(SpannableStringBuilder.java:116)
       at com.tokenautocomplete.TokenCompleteTextView$TokenTextWatcher.afterTextChanged(TokenCompleteTextView.java:1184)
       at android.widget.TextView.sendAfterTextChanged(TextView.java:7694)
       at android.widget.TextView$ChangeWatcher.afterTextChanged(TextView.java:9482)
       at android.text.SpannableStringBuilder.sendAfterTextChanged(SpannableStringBuilder.java:972)
       at android.text.SpannableStringBuilder.replace(SpannableStringBuilder.java:516)

**Comment by mgod** [comment]
I believe I've fixed this in c60d372a45f9197e98a83c7fc5aa7ad1e8d949ce. I expect to have a release with this fix out in the next few days.
